src/parser.rs

```diff
@@ -1195,7 +1206,7 @@ impl Parser {
         let span_start = self.position();
         self.keyword(b"for");
 
-        let variable = self.variable();
+        let variable = self.variable_decl();
```

Comment:
Good call. I personally would favor us consolidating towards the binding without `$` but that can be discussed later.